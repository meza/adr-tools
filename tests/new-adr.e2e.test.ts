/* eslint-disable no-sync */
import { describe, it, expect, afterEach, beforeEach } from 'vitest';
import * as childProcess from 'child_process';
import * as path from 'path';
import * as fs from 'fs';
import * as os from 'os';

describe('New Adrs', () => {
  const adr = path.resolve(path.dirname(__filename), '../src/index.ts');
  const command = `npx tsx ${adr}`;

  let adrDirectory: string;
  let workDir: string;

  beforeEach(() => {
    // @ts-ignore
    process.env.ADR_DATE = '1992-01-12';
    workDir = path.resolve(fs.realpathSync(fs.mkdtempSync(path.join(os.tmpdir(), 'adr-'))));
    adrDirectory = path.join(workDir, 'doc/adr');
  });

  afterEach(() => {
    childProcess.execSync(`rimraf ${workDir}`);
  });

  it('should create a new one normally', () => {
    childProcess.execSync(`${command} init ${adrDirectory}`, { cwd: workDir });
    childProcess.execSync(`${command} new Example ADR`, { cwd: workDir });

    const expectedNewFile: string = path.join(adrDirectory, '0002-example-adr.md');
    expect(fs.existsSync(expectedNewFile)).toBeTruthy();

    const fileContents = fs.readFileSync(expectedNewFile, 'utf8');
    expect(fileContents).toMatchSnapshot();
  });

  it('should create a new one even if no config exists', () => {
    childProcess.execSync(`${command} new Example ADR`, { cwd: workDir });

    const expectedNewFile: string = path.join(adrDirectory, '0001-example-adr.md');
    expect(fs.existsSync(expectedNewFile)).toBeTruthy();

    const fileContents = fs.readFileSync(expectedNewFile, 'utf8');
    expect(fileContents).toMatchSnapshot();
  });

  it('should create a table of contents upon creation', () => {
    childProcess.execSync(`${command} init ${adrDirectory}`, { cwd: workDir });
    childProcess.execSync(`${command} new Example ADR`, { cwd: workDir });

    const expectedNewFile: string = path.join(adrDirectory, 'decisions.md');
    expect(fs.existsSync(expectedNewFile)).toBeTruthy();

    const fileContents = fs.readFileSync(expectedNewFile, 'utf8');
    expect(fileContents).toMatchSnapshot();
  });

});
