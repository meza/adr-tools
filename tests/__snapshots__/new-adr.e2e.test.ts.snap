// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`New Adrs > should create a new one even if no config exists 1`] = `
"# 1. Example ADR

Date: 1992-01-12

## Status

Accepted

## Context

The issue motivating this decision, and any context that influences or constrains the decision.

## Decision

The change that we're proposing or have agreed to implement.

## Consequences

What becomes easier or more difficult to do and any risks introduced by the change that will need to be mitigated.
"
`;

exports[`New Adrs > should create a new one normally 1`] = `
"# 2. Example ADR

Date: 1992-01-12

## Status

Accepted

## Context

The issue motivating this decision, and any context that influences or constrains the decision.

## Decision

The change that we're proposing or have agreed to implement.

## Consequences

What becomes easier or more difficult to do and any risks introduced by the change that will need to be mitigated.
"
`;

exports[`New Adrs > should create a table of contents upon creation 1`] = `
"# Table of Contents

- [1. Record architecture decisions](0001-record-architecture-decisions.md)
- [2. Example ADR](0002-example-adr.md)"
`;
